### PR TITLE
bpo-46129: Rewrite asyncio.locks tests with IsolatedAsyncioTestCase

### PR DESCRIPTION
**Pull request by asvetlov**
https://bugs.python.org/issue46129

### CODE REVIEW

**Comment by asvetlov** [comment]
The issue can be backported to 3.9-3.10, it doesn't add backward compatibility issues

File: Misc/NEWS.d/next/Library/2021-12-19-10-47-24.bpo-46128.Qv3EK1.rst [hunk 1]
```diff
@@ -0,0 +1,2 @@
+Strip :class:`unittest.IsolatedAsyncioTestCase` stack frames from reported
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Library/2021-12-19-10-47-24.bpo-46128.Qv3EK1.rst]: Wrong PR ?

File: Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst [hunk 2]
```diff
@@ -0,0 +1,2 @@
+Rewrite ``asyncio.locks`` tests with
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst:1]:
```suggestion
Rewrite :mod:`asyncio.locks` tests with
```

File: Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
...
+:class:``unittest.IsolatedAsyncioTestCase` usage.
```
Comment (kumaraditya303) [inline Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst]:
```suggestion
:class:`unittest.IsolatedAsyncioTestCase` usage.
```

File: Lib/test/test_asyncio/test_locks.py [hunk 4]
```diff
@@ -58,25 +51,25 @@ def test_lock_doesnt_accept_loop_parameter(self):
             asyncio.BoundedSemaphore,
         ]
 
+        loop = asyncio.new_event_loop()
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_locks.py]: Should not it be `get_running_loop()`?

File: Lib/test/test_asyncio/test_locks.py [hunk 5]
```diff
@@ -178,32 +166,33 @@ async def lockit(name, blocker):
...
         fa.set_result(None)
         tb.cancel()
         self.assertTrue(lock._waiters[0].cancelled())
-        test_utils.run_briefly(self.loop)
+        await asyncio.sleep(0)
         self.assertFalse(lock.locked())
         self.assertTrue(ta.done())
         self.assertTrue(tb.cancelled())
-        self.assertTrue(tc.done())
+        await tc
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_locks.py:188]: It differs from the former code. Why?

File: Lib/test/test_asyncio/test_locks.py [hunk 6]
```diff
@@ -215,27 +204,23 @@ async def lockit():
...
-        self.assertTrue(t0.done())
+        # Start scheduled tasks
+        await asyncio.sleep(0)
 
-        # Second loop calls trigger
-        test_utils.run_briefly(self.loop)
-        # Third loop calls cancellation
-        test_utils.run_briefly(self.loop)
+        loop.call_soon(trigger)
+        with suppress(asyncio.CancelledError):
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_locks.py]: Maybe `self.assertRaises(asyncio.CancelledError)`?

File: Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst (hunk 2 above)
Comment (asvetlov) [reply Misc/NEWS.d/next/Tests/2021-12-19-12-20-57.bpo-46129.I3MunH.rst:1]: `asyncio.locks` is a *private* undocumented module, user code should import everything from `asyncio` top-level package. The module doesn't exist from Sphinx perspective.

File: Misc/NEWS.d/next/Library/2021-12-19-10-47-24.bpo-46128.Qv3EK1.rst (hunk 1 above)
Comment (asvetlov) [reply Misc/NEWS.d/next/Library/2021-12-19-10-47-24.bpo-46128.Qv3EK1.rst]: Wrong branch merge.
Deleted, thanks.

File: Lib/test/test_asyncio/test_locks.py [hunk 7]
```diff
@@ -668,37 +637,37 @@ async def c2(result):
...
-        test_utils.run_briefly(self.loop)
+        await asyncio.sleep(0)
         self.assertEqual([1, 2], result)
 
         self.assertTrue(t1.done())
         self.assertTrue(t1.result())
         self.assertTrue(t2.done())
         self.assertTrue(t2.result())
 
-    def test_notify_unacquired(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_locks.py]: No need to make this test async.

File: Lib/test/test_asyncio/test_locks.py [hunk 8]
```diff
@@ -668,37 +637,37 @@ async def c2(result):
...
+    async def test_notify_unacquired(self):
         cond = asyncio.Condition()
         self.assertRaises(RuntimeError, cond.notify)
 
-    def test_notify_all_unacquired(self):
+    async def test_notify_all_unacquired(self):
```
Comment (serhiy-storchaka) [inline Lib/test/test_asyncio/test_locks.py]: No need to make this test async.

**Review by serhiy-storchaka — COMMENTED** [review]
LGTM in general, but I have some questions.

File: Lib/test/test_asyncio/test_locks.py (hunk 4 above)
Comment (asvetlov) [reply Lib/test/test_asyncio/test_locks.py]: You are right, any loop (and even non-loop) can be used by this test.

File: Lib/test/test_asyncio/test_locks.py (hunk 5 above)
Comment (asvetlov) [reply Lib/test/test_asyncio/test_locks.py:188]: Technically the same, but the new version also checks that `tc` task was not cancelled.

I can revert if you insist.

File: Lib/test/test_asyncio/test_locks.py (hunk 6 above)
Comment (asvetlov) [reply Lib/test/test_asyncio/test_locks.py]: Done

File: Lib/test/test_asyncio/test_locks.py (hunk 7 above)
Comment (asvetlov) [reply Lib/test/test_asyncio/test_locks.py]: Done

File: Lib/test/test_asyncio/test_locks.py (hunk 8 above)
Comment (asvetlov) [reply Lib/test/test_asyncio/test_locks.py]: Done

**Review by asvetlov — COMMENTED** [review]
@serhiy-storchaka thanks for the review.
Comments are addressed.

File: Lib/test/test_asyncio/test_locks.py (hunk 5 above)
Comment (serhiy-storchaka) [reply Lib/test/test_asyncio/test_locks.py:188]: If it is the same it LGTM.

**Review by serhiy-storchaka — APPROVED** [review]

**Pull request merged**